commit/merkleroot/rmn/client.go

```diff
@@ -19,13 +33,533 @@ type Client interface {
 	ComputeReportSignatures(
 		ctx context.Context,
 		destChain *rmnpb.LaneDest,
-		requestedUpdates []rmnpb.FixedDestLaneUpdateRequest,
+		requestedUpdates []*rmnpb.FixedDestLaneUpdateRequest,
 	) (*ReportSignatures, error)
 }
 
 type ReportSignatures struct {
 	// ReportSignatures are the ECDSA signatures for the lane updates for each node.
+	Signatures []*rmnpb.EcdsaSignature
+	// LaneUpdates are the lane updates for each source chain that we got the signatures for.
 	// NOTE: A signature[i] corresponds to the whole LaneUpdates slice and NOT LaneUpdates[i].
-	Signatures  []rmnpb.EcdsaSignature
-	LaneUpdates []rmnpb.FixedDestLaneUpdate
+	LaneUpdates []*rmnpb.FixedDestLaneUpdate
+}
+
+// PBClient is the base RMN Client implementation.
+type PBClient struct {
+	lggr                                    logger.Logger
+	rawRmnClient                            RawRmnClient
+	rmnNodes                                []RMNNodeInfo
```

Comment:
It is critical that these nodes are in the same order in this slice as the RMN home config right? Maybe add a comment about that?

Reply:
Not required, there is a `NodeIDX` field in RMNNodeInfo.